### DISCUSSION TITLE
Perform multiple reads/searches with snapshot isolation / pausing tasks

### DISCUSSION THREAD

**Discussion by underflow00** [meilisearch/product#798] [Feedback & Feature Proposal]
I sometimes want to perform multiple reads/searches with snapshot isolation (like in an SQL database), i.e. without seeing any changes to the documents made after the first read. It looks like this is not possible currently. Using locks/mutexes to block writes at the application/user level while the reads are also being performed also doesn't work, as the enqueued tasks are still running in the background.

Some form of SQL-like transactions or the ability to pause tasks would be helpful here.

**Comment by macraig** [comment]
Hi @underflow00 , as a bit of context, federated search (but not regular multi-search) guarantees that any index is only opened once per request, preserving a consistent view of the data. We could explore extending this behavior to regular multi-search. Would that help with your use case?

If that wouldn’t fully address your needs, could you share more details on why snapshot isolation is essential for your workflow? While long-lived transactions are currently out of our scope, understanding your specific requirements could help us consider alternative solutions.

**Reply by underflow00** [reply]
Hi. My use case is to perform a search to get a list of documents, then figure out where (at what array index) another arbitrary document (not in meilisearch) fits into that list. The sorted attributes cannot all be retrieved into memory as they may be large, and so multiple passes are needed.

The lack of this feature, and also [this one](https://github.com/meilisearch/meilisearch/discussions/5539) makes implementing this functionality difficult.

If there are workarounds/alternatives, please let me know.
